### PR TITLE
New User Registration Template | OKTA-377085

### PR DESCRIPTION
**Pull request by oleksandrkysil-okta**
Jira ticket:
- https://oktainc.atlassian.net/browse/OKTA-377085

Description:
- New template review

Note:
- It Pending for a youtube video link in the workflows.json file, but the doc files can be reviewed without that.

### CODE REVIEW

File: workflows/new_user_registration/readme.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+      2. Copy the “Invoke URL” from the top of the popup up, but NOT including “?clientToken=&lt;xxxxxxxx". Save this in a notepad.
+      3. Copy the “Client Token”. Save this in a notepad.
+      4. These values shall be used by the calling party to initiate the operation.
+   3. Make sure a connection is selected for instance of the "Okta | Create User" card. If it was not selected automatically, select the necessary connection manually.
+2. Select flow: "[O365] Send Email HTML".
+    1. Select Connection for "Office 365 Mail" card if it was not selected automatically.
+3. Select flow: “[child] process USA customer”
+    1. Make sure a connection is selected for the instance of the “HTTP | Raw Request” card. This card is used to link to the external preference management system.
+4. Select flow: “[child] process CANADA customer”
+    1. Make sure a connection is selected for the instance of the “HTTP | Raw Request” card. This card is used to link to the external preference management system.
```
Comment (davidroossien-okta) [inline workflows/new_user_registration/readme.md:39]: "to a mock external preference management system."

File: workflows/new_user_registration/readme.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+   2. In the Action event card titled “On Demand | API Endpoint”  at the far left of the flow, click the “&lt;/>” (Endpoint settings) link at the bottom right of the card.
+      1. Check the radio button "Expose as Webhook”
+      2. Copy the “Invoke URL” from the top of the popup up, but NOT including “?clientToken=&lt;xxxxxxxx". Save this in a notepad.
+      3. Copy the “Client Token”. Save this in a notepad.
+      4. These values shall be used by the calling party to initiate the operation.
+   3. Make sure a connection is selected for instance of the "Okta | Create User" card. If it was not selected automatically, select the necessary connection manually.
+2. Select flow: "[O365] Send Email HTML".
+    1. Select Connection for "Office 365 Mail" card if it was not selected automatically.
+3. Select flow: “[child] process USA customer”
+    1. Make sure a connection is selected for the instance of the “HTTP | Raw Request” card. This card is used to link to the external preference management system.
```
Comment (davidroossien-okta) [inline workflows/new_user_registration/readme.md:37]: "to a mock external preference management system."

File: workflows/new_user_registration/workflow.json [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    "okta","office365mail"
+  ],
+  "details": {
+    "flowCount": 5,
+    "stashCount": 0
+  },
+  "links": [
+    {
+      "title": "Video Walk-through",
+      "destination": ""
```
Comment (davidroossien-okta) [inline workflows/new_user_registration/workflow.json:15]: missing link

**Review by davidroossien-okta — CHANGES_REQUESTED** [review]

**Review by markmoussa-okta — COMMENTED** [review]
@davidroossien-okta I added some edits to both the summary in the workflow.json and the introductory content in readme. I'll ping you offline to streamline the review process for all such non-InfoDev-sourced documentation.

File: workflows/new_user_registration/readme.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
+# **New User Registration** 
+
+
+## Overview
+
+In CIAM use cases, many business units, locales, and brands may require distinct user management operations. Okta Workflows can help implement custom processing of the registration context. 
+
+In this template, the context of new user registration is processed by Workflows which shall customize the branding, customize the birthright enablement, and link to external systems for duplication verification and Preference Management.
```
[the remark this answers is not in the document]
Comment (markmoussa-okta) [reply workflows/new_user_registration/readme.md]: @davidroossien-okta Beats me. What about "Preference Management" - is that a proper noun? There's some oddities here that I need to clear up with you.

File: workflows/new_user_registration/workflow.json [hunk 5]
```diff
@@ -0,0 +1,22 @@
+{
+  "name": "new_user_registration",
+  "title": "New User Registration",
+  "description": "In CIAM use cases, many business units, locales, and brands may require distinct user management operations. Okta Workflows can help implement custom processing of the registration context.",
```
[the remark this answers is not in the document]
Comment (oleksandrkysil-okta) [reply workflows/new_user_registration/workflow.json]: Updated

File: workflows/new_user_registration/readme.md [hunk 6]
```diff
@@ -0,0 +1,85 @@
+# **New User Registration** 
+
+
+## Overview
+
+In CIAM use cases, many business units, locales, and brands may require distinct user management operations. Okta Workflows can help implement custom processing of the registration context.
```
[the remark this answers is not in the document]
Comment (oleksandrkysil-okta) [reply workflows/new_user_registration/readme.md]: This was removed, so no need in changes

File: workflows/new_user_registration/readme.md [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+## Overview
+
+In CIAM use cases, many business units, locales, and brands may require distinct user management operations. Okta Workflows can help implement custom processing of the registration context. 
+
+In this template, the context of new user registration is processed by Workflows which shall customize the branding, customize the birthright enablement, and link to external systems for duplication verification and Preference Management.
+The External Check for existing users is implemented with an API stub. In an actual use case, a specific HTTP Connector shall be configured along with the needed API Request JSON to match your system requirements. Similarly, the Preference Management 
+is implemented with a stub, but can be substituted with an Okta Workflow built-in Connector such as DataGrail or OneTrust, as well as, any system that supports API communication.
+
+
+## Before you get Started / Prerequisites
```
[the remark this answers is not in the document]
Comment (oleksandrkysil-okta) [reply workflows/new_user_registration/readme.md]: removed

File: workflows/new_user_registration/readme.md [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+    1.  Make sure a connection is selected for the instances of the “Okta” cards.If it was not selected automatically, select the necessary connection manually.
+6. Ensure all these flows are turned on:
+    * Create_User_wExtCheck
+    * [O365] Send Email HTML
+    * [child] process USA customer
+    * [child] process CANADA customer
+    * [child] Add User to Okta GroupName
+
+
+## Testing this Flow
```
[the remark this answers is not in the document]
Comment (oleksandrkysil-okta) [reply workflows/new_user_registration/readme.md]: Changed

File: workflows/new_user_registration/readme.md [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+
+2. Hit the “Send” for the POSTMAN request.
+3. View Flow History and verify it was completed successfully.
+4. Go to the Okta tenant Admin UI-> People.
+5. Verify the test user exists with correct Group memberships and profile attributes.
+
+
+## Limitations & Known Issues
+
+*   None
```
[the remark this answers is not in the document]
Comment (oleksandrkysil-okta) [reply workflows/new_user_registration/readme.md]: Changed

File: workflows/new_user_registration/readme.md [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+*   Email service, in this sample Office 365 Mail is used. Configure Office 365 Mail Workflow Connector
+*   Create Okta group named "AcmeUsers"
+*   The External API calls in this sample are using a mock API. Create HTTP Connector for mock API Auth = none
+
+
+## Setup Steps
+
+
+1. Select flow: “Create_User_wExtCheck”
+   1. Make sure a connection is selected for the instance of the “HTTP | Raw Request” card. This card is used to link to an external duplicate user check.
```
[the remark this answers is not in the document]
Comment (oleksandrkysil-okta) [reply workflows/new_user_registration/readme.md]: Fixed

File: workflows/new_user_registration/readme.md (hunk 1 above)
Comment (oleksandrkysil-okta) [reply workflows/new_user_registration/readme.md]: Fixed

File: workflows/new_user_registration/workflow.json (hunk 3 above)
Comment (oleksandrkysil-okta) [reply workflows/new_user_registration/workflow.json]: was added

**Review by davidroossien-okta — APPROVED** [review]

**Pull request merged**